src/query/atomWithQuery.ts

```diff
@@ -132,8 +133,10 @@ export function atomWithQuery<
       const options =
         typeof createQuery === 'function' ? createQuery(get) : createQuery
       const defaultedOptions = queryClient.defaultQueryOptions(options)
-      const observer = get(observerAtom)
-      observer.destroy()
+      const observer = get(
+        queryObserverFamily([queryDataAtom, options.queryKey, getQueryClient])
+      )
+      // observer.destroy()
```

Comment:
This seems really unexpected. We can't get status only with the key? In that case, we need to re-ideate from the ground.